python/lsst/ctrl/oods/butlerAttendant.py

```diff
@@ -100,6 +104,33 @@ def extract_info_val(self, dataId, key, prefix):
             return f"{prefix}{dataId[key]:02d}"
         return f"{prefix}??"
 
+    def _filter_files(self, resource_paths: list, patterns: list):
+        """Filters filenames based on a list of patterns
+
+        Parameters
+        ----------
+        resource_paths: `list`
+            A list of ResourcePath
+
+        Returns:
+        -------
+        tuple: `list`, `list`
+               - matching_files (list): names containing at least one pattern
+               - non_matching_files (list): names containing no patterns
+        """
```

Comment:
* Can we include the type specification for the lists elements in the function signature and the docstring as well? (E.g. change`resource_path: list` -> `resource_path: list[ResourcePath]`)
* The description of the `patterns` parameter is not included in the docstring.
* In the docstring, please add a single space _before_ the  colons in the descriptions of 
* Please remove the colon after `Returns` (line 115).
* I'd rewrote the description of the return values as:
```
matching_files : `list` [`lsst.resources.ResourcePath`]
    Paths matching at least one pattern.
non_matching_files :  `list` [`lsst.resources.ResourcePath`]
    Paths matching no pattern.
```

Reply:
Thanks.  Fixed.